classes/local/skel/lib_php_file.php

```diff
@@ -66,4 +66,26 @@ public function add_supported_feature($feature) {
 
         $this->data['self']['supports'][] = $feature;
     }
+
+    /**
+     * Returns the variables needed for rendering the template.
+     *
+     * @param string $plugintype.
+     * @return string[].
+     */
+    static public function get_template_variables($plugintype = null) {
+
+        $templatevars = array();
+
+        if ($plugintype == 'enrol') {
+            $templatevars = array(
+                array('name' => 'allow_enrol', 'hint' => 'boolean', 'values' => array('True', 'False')),
```

Comment:
Hmm. It sounds like mixing various concepts into this single hint field. Partially you describe it as a variable type (boolean, integer), partially as a validation flag (strict, warning). And the name "hint" itself suggests that it is some sort of inline help to be displayed to the user in the UI.

This usage of values does not sounds right to me, too. For booleans, it makes no sense to have to explicitly declare true / false as possible values, right? For integers, surely you would not enumerate all possible numbers ;-)

I would suggest to have these things designed in a cleaner and more intuitive way.  Let me encourage you to re-think this again.